### ISSUE TITLE
Test distribution tarball

### ISSUE THREAD

**Issue by jdinan** [Sandia-OpenSHMEM/SOS#647]
Test that the distribution tarball works, as follows:
```
$ ./configure --with-ofi=$HOME/opt/libfabric --enable-pmi-simple
$ make dist
$ tar xvzf sos-1.4.1.tar.gz
$ cd sos-1.4.1
$ ./autogen.sh
$ mkdir build
$ cd build
$ ../configure.sh [config args]
$ make check
```

**Comment by wrrobin** [comment]
Checked. Everything is ok except there is no `autogen.sh` in the tarball. It already has the `configure` and other scripts built.

**Comment by wrrobin** [comment]
Checked and closed by #656
